contributingGuides/TYPESCRIPT_STYLE.md

```diff
@@ -0,0 +1,857 @@
+
+Usage of `any` is forbidden unless there is a very good reason for it. The `any` type bypass TypeScript type-checking system, thus making the code less safe and more prone to bugs. Any usage of `any` type must be followed with a comment explaining why you are using it.
+
+```ts
+// Bad
+function processAPIResponse(response: any) {
+    // your code
+}
+
+// Good
+// As we are just going to persist the response in Onyx and
+// it would be very hard to type all possible data structures,
+// we just leave `response` as `any`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function processAPIResponse (response: any) {
+    // your code
+}
+```
+
+## TypeScript Annotations
+
+### Usage of `@ts-expect-error`
+
+During migration, it is expected to find TS errors revealing underlying JS issues. Treat significant errors as separate issues linked to the TS migration issue, unless the fix is minor and verifiable (document it in the PR). Use `@ts-expect-error` for such cases.
+
+### Usage of `@ts-ignore`
+
+`@ts-ignore` should not be used. If the usage of `@ts-ignore` cannot be avoided for rare situations, explain the reason in a comment.
+
+## Optional chaining and nullish coalescing
+
+Avoid utilizing `lodashGet` in the future. When working with TypeScript files, opt for the combination of optional chaining (`?.`) and nullish coalescing (`??`) rather than using `lodashGet`.
+
+```ts
+// Bad
+const name = lodashGet(user, 'name', 'Unknown');
+
+// Good
+const name = user?.name ?? 'Unknown';
+```
+
+## Type inference
+
+Code may rely on type inference as implemented by the TypeScript compiler for all type expressions (variables, fields, return types, etc).
+
+```ts
+// Bad
+const name: string = 'name';
+const [counter, setCounter] = useState<number>(0);
+
+// Good
+const name = 'name';
+const [counter, setCounter] = useState(0);
+```
+
+### Return types
+
+For function return types, depending on the clarity of the function use inferred types. Provide explicit typing when it enhances code readability.
+
+```tsx
+// Bad
+function Component() {
+  return (
+    <TextInput onChangeText={(newTextValue: string) => setTextValue(newTextValue)} />
+  );
+}
+
+// Good
+function Component() {
+  return (
+    <TextInput onChangeText={(newTextValue) => setTextValue(newTextValue)} />
+  );
+}
+```
+
+Benefits of explicitly typing return values:
+- More precise documentation to benefit readers of the code.
+- Surface potential type errors faster in the future if there are code changes that change the return type of the function.
+
+```ts
+interface User {
+  id: number;
+  name: string;
+  age: number;
+  active: boolean;
+}
+
+// Bad
+const getUsersSummary = (users: User[], minAge: number, isActive: boolean) => {
+  return users
+    .filter(user => user.age >= minAge && user.active === isActive)
+    .map(user => `${user.name} (${user.age})`);
+};
+
+// Good
+const getUsersSummary = (users: User[], minAge: number, isActive: boolean): string[] => {
```

Comment:
Here we have `string[]` instead of `Array<string>`. I think `string[]` is better.